src/os/bluestore/BlueStore.h

```diff
@@ -805,13 +805,37 @@ class BlueStore : public ObjectStore,
 	used_in_blob.encode(p);
       }
     }
+    template <bool decode_csum>
```

Comment:
Not sure I understand why we need to parametrize decode via templates here. And what's the conceptual difference between 'decode_csum' and 'include_ref_map' parameters which makes their usage that different: regular param vs. templated one?